File: 11900 Boiled Eggs/solution.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return *(int*)a > *(int*)b;
}

int main() {
    int T;
    scanf("%d", &T);
    for(int i = 1; i <= T; ++i) {
        int n, P, Q;
        scanf("%d %d %d", &n, &P, &Q);
        int weights[n];
        for(int j = 0; j < n; ++j)
            scanf("%d", &weights[j]);
        qsort(weights, n, sizeof(int), compare);
        int total = 0, count = 0;
        n = n < P ? n : P;
        while(count < n) {
            if(total + weights[count] > Q)
                break;
            total += weights[count++];
        }
        printf("Case %d: %d\n", i, count);
    }
	return 0;
}

File: 11900 Boiled Eggs/solution.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
		st.nextToken();
		int T = (int)st.nval;
		StringBuilder output = new StringBuilder();
		for(int i = 1; i <= T; ++i) {
		    st.nextToken();
		    int n = (int)st.nval;
		    st.nextToken();
		    int P = (int)st.nval;
		    st.nextToken();
		    int Q = (int)st.nval;
		    int[] weights = new int[n];
		    for(int j = 0; j < n; ++j) {
		        st.nextToken();
		        weights[j] = (int)st.nval;
		    }
            Arrays.sort(weights);
            int total = 0, count = 0;
            n = Math.min(n, P);
            while(count < n) {
                if(total + weights[count] > Q)
                    break;
                total += weights[count++];
            }
            output.append("Case ").append(i).append(": ").append(count).append('\n');
        }
		System.out.print(output);
	}
}
